File: fullstack-mobile-backend-system-complete/backend/src/utils/logger.ts
import winston from 'winston';
import { LOG_LEVEL } from '../config/config';

const { combine, timestamp, printf, colorize, align } = winston.format;

const logger = winston.createLogger({
  level: LOG_LEVEL,
  format: combine(
    colorize({ all: true }),
    timestamp({
      format: 'YYYY-MM-DD hh:mm:ss A',
    }),
    align(),
    printf((info) => `[${info.timestamp}] ${info.level}: ${info.message}`)
  ),
  transports: [
    new winston.transports.Console(),
    // new winston.transports.File({ filename: 'logs/error.log', level: 'error' }),
    // new winston.transports.File({ filename: 'logs/combined.log' }),
  ],
});

export default logger;

File: production-web-scraper-backend-pro/backend/core/logger.py
```python
import sys
from loguru import logger
from backend.core.config import settings

logger.remove()
logger.add(
    sys.stderr,
    level=settings.LOG_LEVEL,
    colorize=True,
    format="<green>{time:YYYY-MM-DD HH:mm:ss}</green> | <level>{level: <8}</level> | <cyan>{name}</cyan>:<cyan>{function}</cyan>:<cyan>{line}</cyan> - <level>{message}</level>"
)

# Optional: Add a file sink for production environments
# logger.add(
#     "logs/app.log",
#     rotation="5 MB",
#     retention="10 days",
#     level=settings.LOG_LEVEL,
#     format="{time:YYYY-MM-DD HH:mm:ss.SSS} | {level: <8} | {name}:{function}:{line} - {message}"
# )
```